dace/transformation/dataflow/otf_map_fusion.py

```diff
@@ -50,44 +49,93 @@ def can_be_applied(self, graph, expr_index, sdfg, permissive=False):
                 if prod != self.first_map_exit:
                     return False
 
-        # No WCR on first map
-        for edge in graph.in_edges(self.first_map_exit):
-            if edge.data.wcr is not None:
-                return False
+        # Equation Solvability: First map must induce a non-ambiguous set of writes for all possible reads of second map.
+
+        # dims(first map) <= dims(second map)
+        first_map_entry = graph.entry_node(self.first_map_exit)
+        if len(first_map_entry.map.params) > len(self.second_map_entry.map.params):
+            return False
 
-        # Only single write per array
-        write_memlets = {}
+        # Write memlets are unique
+        out_memlets = {}
         for edge in graph.in_edges(self.first_map_exit):
             memlet = edge.data
-            array = memlet.data
 
-            if array in write_memlets:
-                return False
+            # Limitation: Only scalars for now
+            for access in memlet.subset:
+                b, e, s = access
+                if s > 1:
+                    return False
 
-            write_memlets[array] = memlet
+                length = symbolic.evaluate(e - b, symbols=sdfg.constants)
+                if length > 1:
+                    return False
 
-        # Memlets access must be inferable
-        first_map_entry = graph.entry_node(self.first_map_exit)
+            # WCR must have identity zero. Otherwise, we must specifically initialize the temporary data
```

Comment:
that is also the case for zero identity, but you use setzero. Please note in the code that you use `setzero` below.